developing/code-reviews/pull-request-best-practices.md

```diff
@@ -0,0 +1,25 @@
+# Best Practices for Submitting a Pull Request
+
+Generally, a best practice to engage in before submitting your own pull request is to do a code review on your work.
+
+Prior to submitting a PR, review your code for the following characteristics:
+
+1. Is the code **READABLE**?
+   - Is it well-organized?
+   - Are the class, variable, and method names clear and descriptive?
+   - Are the formatting, spacing, comments, and indentations consistent?
+   - Have you refactored the code to reduce repetition and nested code?
+   - Have you included the appropriate comments to make your code readable to others?
+
+2. Have you followed the language’s **STYLE CONVENTIONS**?
+   - Are class, variable, and method names cased and capitalized according to the style conventions of the language (e.g. camelCase vs. snake_case)
+
+3. Is the code **CORRECT**?
+   - Does the code work as intended?
```

Comment:
It might be good to add something here about instructions for the reviewer to be able to test the code themselves. I've seen bugs get uncovered this way, assumptions questioned, etc. and it's helpful to the reviewer's learning as well.

Reply:
Do you mean the author should be writing tests or describing how to test it?  Or not to breeze through a pull request and try checking out the code and test that it works as intended?

Reply:
Both I guess, but more the latter. I assume that the author writing tests is an implicit expectation (depending on the PR), but that the author should also specify how to run (if applicable) or otherwise test the thing that the PR is intended to be doing.